webapp/src/js/enketo/widget-utils.js

```diff
@@ -0,0 +1,57 @@
+const moment = require('moment');
+
+
+const getBindFunction = (widgetName, widgetClass) => {
+  return function (options, event) {
+    return this.each(function () {
+      const $this = $(this);
+      let data = $this.data(widgetName);
+
+      options = options || {};
+
+      if (!data && typeof options === 'object') {
+        $this.data(widgetName, (data = new widgetClass(this, options, event)));
+
+      } else if (data && typeof options === 'string') {
+        data[options](this);
+      }
+    });
+  };
+};
+
+const getForm = () => {
+  return window.CHTCore.Enketo.getCurrentForm();
+};
+
+const setFieldValue = ($widget, fieldName, value) => {
+  if (!fieldName || value === undefined) {
+    return;
+  }
+
+  $widget
+    .find(`input[name$=${fieldName}]`)
+    .val(value)
+    .trigger('change');
+};
+
+const getFieldValue = (form, fieldName) => {
+  if (!form || !fieldName) {
+    return;
+  }
+
+  return form.model.$
+    .find(fieldName)
```

Comment:
So, this will find any field that matches the name? what if there are multiple fields that match this name? Is the first one picked to get text or? maybe we should make this search more exact, so we can never match more than one field?

Reply:
This `find()` is a bit different from usual, it does a find over Enketo model (enketo-core) which looks like this: 

<img width="414" alt="Screen Shot 2021-05-24 at 5 18 01 pm" src="https://user-images.githubusercontent.com/66472237/119337413-fe37b480-bcb8-11eb-9d49-94b8eea368b9.png">

Where it matches using exact declared field and regular DOM matches won't be supported at this level:

<img width="708" alt="Screen Shot 2021-05-24 at 5 08 31 pm" src="https://user-images.githubusercontent.com/66472237/119337542-27584500-bcb9-11eb-807f-ab1bc83a93a2.png">

Reply:
I added comments in each function explaining what is the "form" parameter and details

Reply:
I've changed the parameter name to "fieldSelector", because people might want to access fields in <context>, or <session> or <data> and make the match specific.

Reply:
Just coming back to this as Im curious how we handle the form having multiple patient_id fields, which could happen. Was that addressed somewhere else already?

Reply:
Hi @abbyad, I tried to find a form in core containing more than one field of patient_id, but I guess since these are reserved fields then they are best defined at root level 🤔 
Do you remember a form so I can take as example for this case? Are these multiple patient_id fields referring to different patients for the Rapid Diagnosis Test reports?

Reply:
There are forms with copies of the patient information but these fields have different name and receives the value from the original by using calculation

Reply:
We may not have such a form in our default set of forms but the `patient_id` is not a reserved unique field per form. For instance the form inputs can have the `_id` of multiple contacts in the hierarchy. It would be possible that the `patient_id` for multiple of those contacts to be included in the form.

It would help to know how the rdtoolkit widgets are defined in XLSForm, and what the requirements are (if any) with associated fields. I couldn't find that in the [doc](https://docs.google.com/document/d/1efEQnqHMZvC_I44_26GlVsQ3fnjdwcKhaTVw2wj6G6o/edit), or a PR for the [docs site](https://docs.communityhealthtoolkit.org/apps/reference/forms/app/#cht-xform-widgets).

Reply:
Oh so we can have multiple patient info in one form. We need the one coming from the contact search field (source user), so taking the root's patient might be fine 🤔 as long as the selector takes the immediate child of root (root > field)
I'm going to try to put another patient in deeper level of the XLS and see what happens (I hope that's valid test 😅)

The XLS are in this PR but easier to see in [this shared folder](https://drive.google.com/drive/folders/1lqmn1QXlLTYfN_Tt1zF93_R6hDocdfvg?usp=sharing). There's also [this document about setup](https://docs.google.com/document/d/1vREvxhQv2jmSBOzLKcebxk49gMCO9jjqnb8UR_DWKo4/edit?usp=sharing) but needs to be updated (after solution is approved) with the latest shape of XLS but the general idea is there. 
I'm going to send the documentation to the cht-docs :)

Reply:
I've completed the test and the widget is working fine since we are using `>` operator that refers to the immediate child on the level, our selectors for patient: `[form root] > patient_id` and `[form root] > patient_name`

I tested by creating a form type having `other_contact` as the example :)  and using webapp to fill and save the information as regular chw user:
<img width="500" alt="Screen Shot 2021-06-01 at 11 53 47 am" src="https://user-images.githubusercontent.com/66472237/120265588-ed97c780-c2d2-11eb-85d9-887cf5cebca0.png">

<img width="500" alt="Screen Shot 2021-06-01 at 12 15 17 pm" src="https://user-images.githubusercontent.com/66472237/120265734-2df74580-c2d3-11eb-8771-6a441d4b9259.png">

Additionally exploring with console:
<img width="500" alt="Screen Shot 2021-06-01 at 12 03 42 pm" src="https://user-images.githubusercontent.com/66472237/120265615-f2f51200-c2d2-11eb-93a0-b461f8031b2a.png">

Reply:
Thanks for the details!